ui/app/services/data-caches.js

```diff
@@ -0,0 +1,33 @@
+import Service, { inject as service } from '@ember/service';
+
+export const COLLECTION_CACHE_DURATION = 60000; // one minute
+
+export default class DataCachesService extends Service {
+  @service router;
+  @service store;
+  @service system;
+
+  collectionLastFetched = {};
+
+  async fetch(modelName) {
+    const modelNameToRoute = {
```

Comment:
Maybe this could be dynamically calculated by examining routes with watchers but that seemed like overkill for this small use.

Reply:
I agree. This quick little mapping does the job just fine.